src/RtsClient.js

```diff
@@ -1,5 +1,25 @@
 const q = require('q');
-const request = q.nfbind(require('browser-request'));
+const request = (opts) => {
+    const expectingJSONOnSucess = opts.json;
+    if (opts.json) {
+        opts.json = false;
+    }
+    return q.nfbind(require('browser-request'))(opts).then((args) => {
+        const response = args[0];
+        let body = args[1];
+
+        // Do not expect JSON on error status code, throw error instead
+        if (response.statusCode !== 200) {
+            throw new Error(body);
+        }
+
+        if (expectingJSONOnSucess) {
+            body = JSON.parse(body);
+        }
+
+        return [response, body];
+    });
+};
```

Comment:
Also I'm not super keen on making `request` be something that is a wrapper around browser-request with more functionality, so I'd say if you're making it behave different on non 2xx responses, that should be a separate function. (Also why does it needs to explicitly parse the json? Doesn't passing the `json` flag through to request do that?)

We're gaining a bit of a collection of parts of react sdk doing requests in slightly different ways (DuckDuckGoProvider.js, MFileBody.js, DecryptFile.js and now here). We should try &make them roughly similar.